activedata_etl/transforms/per_test_to_es.py

```diff
@@ -39,9 +43,16 @@ def create_record(parent_etl, count, filename, covered, uncovered):
             urls_w_uncoverable_lines.add(artifact.url)
             Log.warning("per-test-coverage {{url}} has uncoverable lines", url=artifact.url)
 
+        language = "unknown"
+        for lang, extensions in LANGUAGE_MAPPINGS:
+            for extension in extensions:
+                if filename.endswith("." + extension):
```

Comment:
if the extensions included the dot (`"."`), and were a `tuple`, not a `list` then

    language = [lang for lang, extensions in LANGUAGE_MAPPINGS if filename.endswith(extensions)]

this defines `language` as an array, but Elasticsearch treats a value as an array-of-one-value; making no distinction between `["js"]` and `"js"`.  The code that inserts into ES performs a "scrub" that simplifies the JSON before insertion: https://github.com/klahnakoski/ActiveData-ETL/blob/56ae0ffd56e72031655573f706c9fa3eedd26113/vendor/pyLibrary/env/elasticsearch.py#L1044:L1049

Reply:
Done with 8823d14!